docs/developer-guide/working-with-rules.md

```diff
@@ -1,69 +1,70 @@
+
+Here is the basic format of the source file for a rule:
 
 ```js
 /**
- * @fileoverview Rule to flag use of an empty block statement
+ * @fileoverview Rule to disallow unnecessary semicolons
  * @author Nicholas C. Zakas
  */
+
 "use strict";
 
 //------------------------------------------------------------------------------
 // Rule Definition
 //------------------------------------------------------------------------------
 
 module.exports = function(context) {
-
     return {
-        // properties go here
+        // callback functions
     };
-
 };
 
-module.exports.schema = [
-    // JSON Schema for rule options goes here
-];
+module.exports.schema = []; // no options
 ```
 
-**Important:** Rule submissions will not be accepted unless they are in this format.
-
 ## Rule Basics
 
-Each rule is represented by a single object with several properties. The properties are equivalent to AST node types from [ESTree](https://github.com/estree/estree). For example, if your rule wants to know when an identifier is found in the AST, then add a method called "Identifier", such as:
+`schema` (array) specifies the [options](#options-schemas) so ESLint can prevent invalid [rule configurations](../user-guide/configuring#configuring-rules)
 
-```js
-module.exports = function(context) {
+`create` (function) returns an object with methods that ESLint calls to "visit" nodes while traversing the abstract syntax tree (AST as defined by [ESTree](https://github.com/estree/estree)) of JavaScript code:
 
-    return {
+* if a key is a node type, ESLint calls that **visitor** function while going **down** the tree
+* if a key is a node type plus `:exit`, ESLint calls that **visitor** function while going **up** the tree
+* if a key is an event name, ESLint calls that **handler** function for [code path analysis](./code-path-analysis.md)
 
-        "Identifier": function(node) {
-            // do something with node
-        }
-    };
-
-};
-```
-
-Each method that matches a node in the AST will be passed the corresponding node. You can then evaluate the node and it's surrounding tree to determine whether or not an issue needs reporting.
+A rule can use the current node and its surrounding tree to report or fix problems.
 
-By default, the method matching a node name is called during the traversal when the node is first encountered, on the way down the AST. You can also specify to visit the node on the other side of the traversal, as it comes back up the tree, by adding `:exit` to the end of the node type, such as:
+Here are methods for the `array-callback-return` rule:
 
 ```js
 module.exports = function(context) {
-
+    // declare the state of the rule
     return {
-
-        "Identifier:exit": function(node) {
-            // do something with node
+        ReturnStatement: function(node) {
+            // at a ReturnStatement node while going down
+        },
+        "FunctionExpression:exit":
```

Comment:
@platinumazure You will smile. I woke up this morning thinking how I left out the `case` keywords at the fallthrough for function expressions, and then, oh noooo!